### PR TITLE
Fixes in pyx files

### PR DESCRIPTION
**Pull request by fchapoton**
This is fixing a few issues about division in various cython files.

All the fixes were found using the command `cython -a -3 *.pyx`.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.

### CODE REVIEW

**Comment by fchapoton** [comment]
@kedlaya : could you please have a look at the 2 failing doctests in hypergeometric motives ? I wonder if the old results were correct.

**Comment by roed314** [comment]
I just checked in Magma, which agrees with Sage's old results (note that there's a `t <-> 1/t` difference in convention, thus the `1/2`):
```
H := HypergeometricData([[3],[4]]);
EulerFactor(H,1/2,7);
7*$.1^2 - 3*$.1 + 1
```

Indeed, 21 is too big since it needs to be a weight 1 Weil polynomial.

**Comment by roed314** [comment]
I haven't looked at the code in detail, but I wonder why you even want to change the signature here.  It doesn't seem to relate to the C division convention.

**Comment by fchapoton** [comment]
Thanks. So my changes are not good. I will have to investigate. Probably I will undo the changes here about hypergeometric motives.

**Comment by fchapoton** [comment]
I wanted to fix the warning displayed by "cython -a -3 hypergeometric_misc.pyx".

**Comment by kedlaya** [comment]
Off the top of my head I am guessing that changing `p` to `unsigned long long` is changing the behavior of `%` down the line. This is of course fixable, let me have a look.

**Comment by fchapoton** [comment]
I have tried to make a smaller change in hgm,  and this seems to work. Not sure that it appeases "cython -a -3" buit this is maybe not important

**Comment by kedlaya** [comment]
> I have tried to make a smaller change in hgm, and this seems to work. Not sure that it appeases "cython -a -3" buit this is maybe not important

On my end changing only `f` to be `unsigned int` seems to fix the warning without disturbing the doctest (which makes sense as the issue is with raising an integer to a possibly negative integer power).

**Comment by fchapoton** [comment]
Thanks. So this is now ready for review.

**Comment by fchapoton** [comment]
please review, if you have time

File: src/sage/combinat/designs/subhypergraph_search.pyx [hunk 1]
```diff
@@ -180,7 +180,7 @@ cdef hypergraph h_init(int n,list H) noexcept:
     cdef hypergraph h
     h.n          = n
     h.m          = len(H)
-    h.limbs      = (n+63)/64 # =ceil(n/64)
+    h.limbs      = (n+63)//64 # =ceil(n/64)
```
Comment (dcoudert) [inline src/sage/combinat/designs/subhypergraph_search.pyx]: 2 spaces before #

File: src/sage/schemes/elliptic_curves/mod_sym_num.pyx [hunk 2]
```diff
@@ -456,24 +456,24 @@ cdef int best_proj_point(llong u, llong v, llong N,
     else:  # cases like (p:q) mod p*q drop here
         p = llgcd(u, N)
         q = llgcd(v, N)
-        Nnew = N / p / q
-        w = ((u/p) * llinvmod(v/q, Nnew)) % Nnew
-        y0 = N/q
+        Nnew = N // p // q
```
Comment (dcoudert) [inline src/sage/schemes/elliptic_curves/mod_sym_num.pyx]: I know it's ok, but isn't it better for readability to write `(N // p) // q`?

**Review by dcoudert — CHANGES_REQUESTED** [review]
otherwise LGTM.

**Comment by fchapoton** [comment]
merci ! Voilà !

**Review by dcoudert — APPROVED** [review]
LGTM.

**Pull request merged**